addons/product/models/product_template.py

```diff
@@ -774,6 +788,33 @@ def _get_mapped_attribute_names(self, parent_combination=None):
             for attribute_value in all_product_attribute_values
         }
 
+    def _can_variant_be_used(self, combination):
+        """Return whether the variant can be used based on its own combination.
```

Comment:
This naming of method is not correct, it's unrelated to variants, it's just a generic "combination" method, that does the check in regard to the "config of attributes on the template".